### PR TITLE
Implemented missing methods in BigMap

### PR DESCRIPTION
**Pull request by JulijaRamoskiene**
Closes https://github.com/iTwin/imodel-transformer/issues/134
- Changed BigMap to implement Map instead of extending it. Implemented missing methods.
- Added BigMap tests.
- Added iModelTransformer test to verify that unresolved references are logged correctly. Moved logger initialization logic to beforeEach(), as it is reconfigured in new test case.

### CODE REVIEW

File: packages/transformer/src/BigMap.ts [hunk 1]
```diff
@@ -83,4 +82,36 @@ export class BigMap<V> extends Map<Id64String, V> {
...
+        for (let key of map.keys())
+          yield key;
+      }
+    }
+
+    *values(): IterableIterator<V> {
+      var maps = Object.values(this._maps);       
+      for (var map of maps) {
+        for (let value of map.values())
+          yield value;
```
Comment (MichaelBelousov) [inline packages/transformer/src/BigMap.ts:114]:
```suggestion
      const maps = Object.values(this._maps);       
      for (const map of maps) {
        for (const value of map.values())
          yield value;
```

File: packages/transformer/src/BigMap.ts [hunk 2]
```diff
@@ -83,4 +82,36 @@ export class BigMap<V> extends Map<Id64String, V> {
...
+      for (var map of maps) {
+        for (let [key, value] of map.entries())
+          yield [key, value];
+      }
+    }
+
+    *keys(): IterableIterator<Id64String> {
+      var maps = Object.values(this._maps);       
+      for (var map of maps) {
+        for (let key of map.keys())
```
Comment (MichaelBelousov) [inline packages/transformer/src/BigMap.ts:105]:
```suggestion
      const maps = Object.values(this._maps);       
      for (const map of maps) {
        for (const key of map.keys())
```

File: packages/transformer/src/BigMap.ts [hunk 3]
```diff
@@ -83,4 +82,36 @@ export class BigMap<V> extends Map<Id64String, V> {
...
+    }
+
+    get [Symbol.toStringTag]() {
+      return 'BigMap';
+    }
+
+    *entries(): IterableIterator<[Id64String, V]> {
+      var maps = Object.values(this._maps);       
+      for (var map of maps) {
+        for (let [key, value] of map.entries())
```
Comment (MichaelBelousov) [inline packages/transformer/src/BigMap.ts:97]:
```suggestion
      const maps = Object.values(this._maps);       
      for (const map of maps) {
        for (const [key, value] of map.entries())
```

File: packages/transformer/src/test/standalone/BigMap.test.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import { assert } from "chai";
+
+describe ("BigMap", function () {
+
+    // Test map keys will be assigned into 2 different submaps when BigMap is created
+    const testMap = new Map ([["0x123f", "testVal1"], ["0x1231", "testVal2"]]);
+    
+    const createBigMap = (map: Map<string, string>) : BigMap<string> => {
+        const bigMap = new BigMap <string> ();
+        for (let entry of map.entries ()) {
```
Comment (MichaelBelousov) [inline packages/transformer/src/test/standalone/BigMap.test.ts:16]:
```suggestion
        for (const entry of map.entries ()) {
```

File: packages/transformer/src/test/standalone/BigMap.test.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            assert.deepEqual(actualMap, testMap);
+        });
+    });
+
+    describe ("iterator", function () {
+        it("should get all values", async function () {
+            const bigMap = createBigMap (testMap);
+
+            const actualMap  = new Map();
+            for (let entry of bigMap) {
```
Comment (MichaelBelousov) [inline packages/transformer/src/test/standalone/BigMap.test.ts:49]:
```suggestion
            for (const entry of bigMap) {
```

**Review by MichaelBelousov — COMMENTED** [review]
thank you so much for the tests!
Can you add a test proving that https://github.com/iTwin/imodel-transformer/issues/134 is fixed?

**Comment by JulijaRamoskiene** [comment]
> Great `BigMap` tests! Can you add a test proving that #134 is fixed?

Added "should log unresolved references" test case

File: packages/transformer/src/test/standalone/IModelTransformer.test.ts [hunk 6]
```diff
@@ -642,6 +645,55 @@ describe("IModelTransformer", () => {
...
+    Logger.setLevelDefault(LogLevel.Warning);
+
+    // Act
+    await transformerA2S.processAll();
+
+    // Collect expected ids
+    const result = iModelA.queryEntityIds({ from: "BisCore.Element",
+      where: "Model.Id = :rootId AND ECInstanceId NOT IN (:rootId, :excludedId)",
+      bindings: {rootId: IModel.rootSubjectId, excludedId } });
+    const expectedIds = [...result].map ((x) => `e${x}`);
```
Comment (nick4598) [inline packages/transformer/src/test/standalone/IModelTransformer.test.ts:683]: Can you also assert that there is atleast one expectedId? I don't think the test would pass if there were 0, but just to make that explicit. You can do that when you get write perms to make a branch if you'd like.

Comment (JulijaRamoskiene) [reply packages/transformer/src/test/standalone/IModelTransformer.test.ts:683]: Added assert in new PR: https://github.com/iTwin/imodel-transformer/pull/144. Closing this PR.

**Comment by JulijaRamoskiene** [comment]
Recreated PR (fork PR's have build issues) :https://github.com/iTwin/imodel-transformer/pull/144

**Pull request closed without merging**